backend/search/query.go

```diff
@@ -243,6 +243,9 @@ func doElasticSearch(ctx context.Context, client *elasticsearch.TypedClient, que
 	if query.Type != nil {
 		indexName = typeToIndexMap[*query.Type]
 	}
+	if indexNameParameter != "" {
+		indexName = indexNameParameter
+	}
```

Comment:
The index is set based on what you are queriying. You should not be able to query for episodes and spec to search the season index.
If that is needed for testing then initialization of the client should accept index config.